File: works/tmp/1929.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using static System.Console;
using System.Text;

namespace ConsoleApp
{
    class Program
    {
        static int MAX = 1000010;
        static int[] p = new int[MAX];

        static void FindPrime()
        {
            for (int i = 2; i < MAX; i++) p[i] = i;

            for (int i = 2; i * i < MAX; i++)
            {
                if (p[i] == 0) continue;
                for (int j = i * i; j < MAX; j += i) p[j] = 0;
            }
        }

        static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(Console.OpenStandardInput());
            StringBuilder sb = new StringBuilder();
            string[] _s = sr.ReadLine().Split(' ');
            int n = int.Parse(_s[0]);
            int m = int.Parse(_s[1]);

            FindPrime();

            for (int i = n; i <= m; i++)
            {
                if (p[i] == 0) continue;
                sb.Append(p[i]).Append('\n');
            }
            Write(sb);
        }

    }
}

File: works/tmp/1929.py
MAX = 1000010
p = [0 for _ in range(MAX)]


def FindPrime():
    global p
    for i in range(2, MAX):
        p[i] = i

    d = int(MAX ** 0.5)
    for i in range(2, d):
        if p[i] == 0: continue
        for j in range(i*i, MAX, i):

            p[j] = 0


FindPrime()
n, m = map(int, input().split())
for i in range(n, m+1):
    if p[i] == 0: continue
    print(p[i])

